src/asm/asm.s

```diff
@@ -334,3 +334,15 @@ _x86_64_asm_xsetbv:
     movl  %esi, %eax    # Second param is the low 32-bits
     xsetbv              # Third param (high 32-bits) is already in %edx
     retq
+
+.global _x86_64_asm_write_mxcsr
+.p2align 4
+_x86_64_asm_write_mxcsr:
+    ldmxcsr %rdi
+    retq
+
+.global _x86_64_asm_read_mxcsr
+.p2align 4
+_x86_64_asm_read_mxcsr:
+    stmxcsr %rax
```

Comment:
The same changes regarding memory should be made here and we also need to use `%rdi` instead of `%rax`. `%rax` is the return value, however `stmxcsr` doesn't move the flags into a register that so that we can use it as the return value, it writes them to a memory location, which in our case is given by `%rdi` - the first function parameter. 

We do something similar when executing `sgdt`: https://github.com/rust-osdev/x86_64/blob/08b520ef770e55538e65db11d46932305089b71a/src/asm/asm.s#L129 And
https://github.com/rust-osdev/x86_64/blob/08b520ef770e55538e65db11d46932305089b71a/src/asm/mod.rs#L175

Reply:
Yeah, the key problem was this: I did not have any way to exercise both code paths. Now I've tested both of them by writing a small script that temporarily modifies the source code so that it compiles with external asm. See the script below.